Code/Waveforms.cpp

```diff
@@ -1348,7 +1356,8 @@ GWFrames::Waveform& GWFrames::Waveform::TransformModesToRotatedFrame(const std::
   // Loop through each mode and do the rotation
   {
     int mode=1;
-    for(int l=std::abs(SpinWeight()); l<NModes; ++l) {
+    const int lmin = lm[0][0];
```

Comment:
For some reason when you don't do this and try to use this function on something with lmin < 2 then it breaks down and you get an error saying that the (1,-1) component cannot be found. This fixes the problem, and I thought that the modes were sorted so this wouldn't be dangerous. I can find another work around or just make sure the modes are sorted.

Reply:
That makes it sound like your change is just papering over a separate problem.  Waveforms should be allowed to have data with ell values less than |s|, but those components should all be zero, so we shouldn't bother trying to rotate them, which is why we *should* start with |s|.  On the other hand Waveforms should *not* be allowed to lack data with ell = |s|, so it's good that an error is raised.  The solution to the problem is to figure out why the input data is wrong; it's not to change the function.  Making allowances for incorrect inputs is a recipe for disaster.

Reply:
You're absolutely right, I was being lazy. I'll go and crack this nut. In the meantime I'll push the fixes I have so far.

Reply:
Ah now I remember what was breaking things. There is a function called `SliceOfTimeIndicesWithEll2` upon which the function `TransformToInertialFrame` is called. Using the |s| tells the `FindIndices` function to look for (1,-1) but the only modes that are available are the l=2 modes. The reason I used `lmin=lm[0][0]` was because this function is called in several places and when it is used in ``SliceOfTimeIndicesWithEll2(blah,blah).TransformToInertialFrame()` then I wanted to not touch the waveform's spin weight but allowed the possibillity to only look at the l=2 modes. The appropriate fix I think is to allow an argument that when specified will account for this, or two have two separate functions (but this seems like code duplication).

Reply:
Sweet!  And now because I needed git practice, I've managed to push that change.  (Of course, you might want to check that the code still actually extrapolates correctly.)